src/components/Card.js

```diff
@@ -14,37 +22,38 @@ export default class Card {
 
   generateCard() {
     this._cardElement = this._getTemplate();
+    this._cardHeart = this._cardElement.querySelector(".card__heart");
     this._setEventListeners();
     this._cardElement.querySelector(".card__name").textContent = this.data.name;
     this._cardElement.querySelector(".card__image").src = this.data.link;
     this._cardElement.querySelector(".card__image").alt = this.data.name;
+    if (this.data.isLiked == true) {
+      this._cardHeart.classList.add("card__heart-option-liked");
+    } else {
+      this._cardHeart.classList.remove("card__heart-option-liked");
+    }
 
     return this._cardElement;
   }
 
   _setEventListeners() {
-    const cardHeart = this._cardElement.querySelector(".card__heart");
     const cardDeleteBtn = this._cardElement.querySelector(".card__delete-btn");
-    cardHeart.addEventListener("click", () => {
+    this._cardHeart.addEventListener("click", () => {
+      this._cardHeart.classList.toggle("card__heart-option-liked");
```

Comment:
You need to **toggle likes** only in block `then` after a successful response from the server because if there is an error, the user will not know about that and will see the toggled heart color.